aten/src/ATen/native/vulkan/VulkanOps.cpp

```diff
@@ -259,12 +259,20 @@ VBuffer kernelNCHW_OCHW_repack_O4C4HWi4o4(
 
 VBuffer bufferFromOptionalHostData(
     c10::optional<const float*> data,
-    const uint32_t size) {
+    const uint32_t dataSize,
+    const uint32_t bufferSize) {
```

Comment:
I am not sure I understand why this function accepts bufferSize as a parameter as well when this function is the entity that ends up allocating the buffer itself a few lines below.  Is it because you want to put more data inside this buffer later on and want to make sure you have enough storage?

Reply:
Yes, that is memory alignment. For example, if we want to bind in shader vec4[4],  our buffer should be 4 * 4 * 4 (float size). While we might have only 14 floats.